Framework/DataHandling/test/LoadNexusLogsTest.h

```diff
@@ -351,6 +351,106 @@ class LoadNexusLogsTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(pclogFiltered2->nthValue(2), 7, 1e-5);
   }
 
+  void test_allow_list() {
+    auto testWS = createTestWorkspace();
+
+    std::vector<std::string> allowed = {"proton_charge", "S2HGap", "S2VGap"};
+
+    LoadNexusLogs loader;
+    loader.setChild(true);
+    loader.initialize();
+    loader.setProperty("Workspace", testWS);
+    loader.setPropertyValue("Filename", "LARMOR00003368.nxs");
+    loader.setProperty<std::vector<std::string>>("AllowList", allowed);
+    loader.setPropertyValue("BlockList", "");
+    loader.execute();
+    TS_ASSERT_THROWS_NOTHING(loader.execute());
+
+    // selog versions
+    allowed.push_back("selog_S2HGap");
+    allowed.push_back("selog_S2VGap");
+
+    // extra proton charge properties
+    allowed.push_back("gd_prtn_chrg");
+    allowed.push_back("proton_charge_by_period");
+    allowed.push_back("nperiods");
+
+    // The default logs that are always present:
+    allowed.push_back("start_time");
+    allowed.push_back("end_time");
+    allowed.push_back("run_title");
+
+    auto run = testWS->run();
+    auto properties = run.getProperties();
+
+    TS_ASSERT_EQUALS(properties.size(), allowed.size());
+
+    // Lookup each name in the workspace property list
+    for (const auto &name : allowed) {
+      bool found = false;
+      for (const auto &prop : properties) {
+        if (prop->name() == name) {
+          found = true;
+          break;
+        }
+      }
+      TS_ASSERT_EQUALS(found, true);
+    }
+  }
+
+  void test_block_list() {
+    auto testWS = createTestWorkspace();
+
+    std::vector<std::string> blocked = {"proton_charge", "S2HGap", "S2VGap"};
+
+    LoadNexusLogs loader;
+    loader.setChild(true);
+    loader.initialize();
+    loader.setProperty("Workspace", testWS);
+    loader.setPropertyValue("Filename", "LARMOR00003368.nxs");
+    loader.setPropertyValue("AllowList", "");
+    loader.setProperty<std::vector<std::string>>("BlockList", blocked);
+    loader.execute();
+    TS_ASSERT_THROWS_NOTHING(loader.execute());
+
+    auto run = testWS->run();
+    auto properties = run.getProperties();
+
+    // add 2 to account for selog_ versions of properties
+    TS_ASSERT_EQUALS(properties.size(), 94 - blocked.size() - 2);
+
+    // Lookup each name in the workspace property list
+    for (const auto &name : blocked) {
+      bool found = false;
+      for (const auto &prop : properties) {
+        if (prop->name() == name) {
+          found = true;
+          break;
+        }
+      }
+      TS_ASSERT_EQUALS(found, false);
```

Comment:
similar question as the one for allowed list test above, we might need a second check to make sure the whole test fails at the error.